pkg/account/api/api_key.go

```diff
@@ -773,3 +773,72 @@ func (s *AccountService) UpdateAPIKey(
 
 	return &proto.UpdateAPIKeyResponse{}, nil
 }
+
+func (s *AccountService) UpdateAPIKeyLastUsedAt(
+	ctx context.Context,
+	req *proto.UpdateAPIKeyLastUsedAtRequest,
+) (*proto.UpdateAPIKeyLastUsedAtResponse, error) {
+	localizer := locale.NewLocalizer(ctx)
+	// No need to check roles since this is only allowed to be called internally by other services.
+
+	err := validateUpdateAPIKeyLastUsedAtRequest(req, localizer)
+	if err != nil {
+		s.logger.Error("Invalid UpdateAPIKeyLastUsedAt request",
+			log.FieldsFromIncomingContext(ctx).AddFields(
+				zap.Error(err),
+				zap.String("apiKeyId", req.ApiKeyId),
+				zap.Int64("lastUsedAt", req.LastUsedAt),
+			)...)
+		return nil, err
+	}
+
+	err = s.mysqlClient.RunInTransactionV2(ctx, func(contextWithTx context.Context, _ mysql.Transaction) error {
+		envAPIKey, err := s.accountStorage.GetEnvironmentAPIKey(contextWithTx, req.ApiKeyId)
+		if err != nil {
+			return err
+		}
+		apiKey := &domain.APIKey{
+			APIKey: envAPIKey.ApiKey,
+		}
+		apiKey.UsedAt(req.LastUsedAt)
+		return s.accountStorage.UpdateAPIKey(contextWithTx, apiKey, envAPIKey.Environment.Id)
+	})
+	if err != nil {
+		if errors.Is(err, v2as.ErrAPIKeyNotFound) {
+			dt, err := statusNotFound.WithDetails(&errdetails.LocalizedMessage{
+				Locale:  localizer.GetLocale(),
+				Message: localizer.MustLocalize(locale.NotFoundError),
+			})
+			if err != nil {
+				return nil, statusInternal.Err()
+			}
+			return nil, dt.Err()
+		}
+		s.logger.Error(
+			"Failed to update api key last used at",
+			log.FieldsFromIncomingContext(ctx).AddFields(
+				zap.Error(err),
+				zap.String("id", req.ApiKeyId),
+			)...,
+		)
+		return nil, api.NewGRPCStatus(err).Err()
```

Comment:
There is an inconsistency here.
When the error is not found, the localizer logic is used. Since this API is for internal use only, we don't need to use the localizer. It's also deprecated.
WDYT?

https://github.com/bucketeer-io/bucketeer/pull/2175/files#diff-1236610833a324b8af6d6273437bdeb2fb35012b5ae3b3374760927a8427c204R813-R815

Reply:
Additionally, by passing the `err` without formatting it, an unknown error will be generated.
https://github.com/bucketeer-io/bucketeer/blob/main/pkg/api/api/grpc_status.go#L41

Since we already have the `statusInternal`, let's use it.

Reference: https://github.com/bucketeer-io/bucketeer/blob/main/pkg/account/storage/v2/api_key.go#L94-L103